#src: https://git.alpinelinux.org/aports/tree/community/open-vm-tools?id=6e0f65e51bfcd02c53bc2b78f46ed18af2dd7ea0
#upstream: no

From a7a78d7963d9dacb79fee1433fc092d67ea0748e Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Wed, 18 Nov 2015 09:03:00 +0000
Subject: [PATCH 04/12] Use configure test for struct timespec

Use the configure script to test for struct time spec instead of trying
to keep track of what platforms has it.

Signed-off-by: Natanael Copa <ncopa@alpinelinux.org>
---
 open-vm-tools/configure.ac           | 1 +
 open-vm-tools/lib/include/hgfsUtil.h | 8 +-------
 2 files changed, 2 insertions(+), 7 deletions(-)

diff --git a/open-vm-tools/configure.ac b/open-vm-tools/configure.ac
index 367e4ff8..5dc0be2b 100644
--- a/open-vm-tools/configure.ac
+++ b/open-vm-tools/configure.ac
@@ -1208,6 +1208,7 @@ AC_TYPE_OFF_T
 AC_TYPE_PID_T
 AC_TYPE_SIZE_T
 AC_CHECK_MEMBERS([struct stat.st_rdev])
+AC_CHECK_MEMBERS([struct timespec.tv_sec],[],[],[[#include <time.h>]])
 AC_HEADER_TIME
 AC_STRUCT_TM
 AC_C_VOLATILE
diff --git a/open-vm-tools/lib/include/hgfsUtil.h b/open-vm-tools/lib/include/hgfsUtil.h
index 609f4c00..a3a022d4 100644
--- a/open-vm-tools/lib/include/hgfsUtil.h
+++ b/open-vm-tools/lib/include/hgfsUtil.h
@@ -53,13 +53,7 @@
 #      include <time.h>
 #   endif
 #   include "vm_basic_types.h"
-#   if !defined _STRUCT_TIMESPEC &&   \
-       !defined _TIMESPEC_DECLARED && \
-       !defined __timespec_defined && \
-       !defined sun && \
-       !defined __FreeBSD__ && \
-       !__APPLE__ && \
-       !defined _WIN32
+#   if !defined HAVE_STRUCT_TIMESPEC_TV_SEC
 struct timespec {
    time_t tv_sec;
    long   tv_nsec;
-- 
2.19.1

